src/equations/acoustic_perturbation_2d.jl

```diff
@@ -112,9 +112,9 @@ A constant initial condition where the state variables are zero and the mean flo
 Uses the global mean values from `equations`.
 """
 function initial_condition_constant(x, t, equations::AcousticPerturbationEquations2D)
-    v1_prime = 0.0
-    v2_prime = 0.0
-    p_prime_scaled = 0.0
+    v1_prime = 0.0f0
+    v2_prime = 0.0f0
+    p_prime_scaled = 0.0f0
```

Comment:
Please do not change the initial conditions and source terms (here and below) right now. This is a case where we would need to discuss how to proceed ideally. To be generic, the floating point type should be determined from the types of `x` and `t` - and maybe coefficients coming from the `equations`. On the other hand, initial conditions used in applications are set in the elixirs, so people could adapt them there accordingly based on the types they want to use.